pkg/cassandra/datacenter.go

```diff
@@ -532,6 +532,20 @@ func ValidateConfig(desiredDc, actualDc *cassdcapi.CassandraDatacenter) error {
 		return fmt.Errorf("tried to change num_tokens in an existing datacenter")
 	}
 
+	desiredRacks := desiredDc.GetRacks()
+	actualRacks := actualDc.GetRacks()
+	if len(desiredRacks) < len(actualRacks) {
```

Comment:
It is frustrating that we can't fold this into [this](https://github.com/k8ssandra/k8ssandra-operator/pull/709/files#diff-7e422b847100c3ee477ee61b97c8ad4da71063c1fc698d4e049081a98c89399cR541) logic. Obviously, we'll get a panic due to the index being out of bounds if len(desiredRacks) < len(actualRacks). I wonder whether it would be clearer to `recover` from the panic instead of doing the bounds check up front.

Reply:
That suggestion is 100% non-blocking by the way, just a stylistic thing.

Reply:
No, never do panic + recover. That's worse than coding Java with always ``} catch(NullPointerException e) {``  instead of doing null checks. Go's panics are not exceptions and are not meant to be used this way.

Reply:
Sure, it was just a thought bubble really. Do you have any docs/resources on why this is a bad idea?

Reply:
panic() has different behavior than normal function returns and thus if the caller is not aware (such as when using goroutines), the execution could be impacted in ways not expected by the user. "The convention in the Go libraries is that even when a package uses panic internally, its external API still presents explicit error return values." (https://go.dev/blog/defer-panic-and-recover)

The problem with panic is that it immediately starts to unwind the goroutine stack. If we look from the perf perspective, then I don't think Go's compiled code even uses traps (like Java for exceptions) to catch these situations, so one could take memory / perf impact also. But that's just IIRC and could have changed during years.

Mostly I'd say because it's a thing that Go itself doesn't use in APIs and as such it's not a idiom (with all the caveats) that usually the caller expects. So, unidiomatic code.